sdk/monitor/Azure.Monitor.Ingestion/tests/ErrorTest.cs

```diff
@@ -183,5 +181,88 @@ Task Options_UploadFailed(UploadFailedEventArgs e)
                 return Task.CompletedTask;
             }
         }
+
+        [Test]
+        public void TwoFailuresWithEventHandlerCancellationToken()
+        {
+            LogsIngestionClient client = CreateClient();
+            // set compression to gzip so SDK does not gzip data (assumes already gzipped)
+            LogsIngestionClient.Compression = "gzip";
+            var entries = GenerateEntries(800, Recording.Now.DateTime);
+            entries.Add(new object[] {
+                    new {
+                        Time = Recording.Now.DateTime,
+                        Computer = "Computer" + new string('*', Mb),
+                        AdditionalContext = 1
+                    }
+                });
+            entries.Add(new object[] {
+                    new {
+                        Time = Recording.Now.DateTime,
+                        Computer = "Computer" + new string('!', Mb),
+                        AdditionalContext = 1
+                    }
+                });
+
+            // Make the request
+            UploadLogsOptions options = new UploadLogsOptions();
+            bool isTriggered = false;
+            var cts = new CancellationTokenSource();
+            options.UploadFailedEventHandler += Options_UploadFailed;
+            AggregateException exceptions = Assert.ThrowsAsync<AggregateException>(async () => { await client.UploadAsync(TestEnvironment.DCRImmutableId, TestEnvironment.StreamName, entries, options, cts.Token).ConfigureAwait(false); });
+            Assert.IsTrue(isTriggered);
+            Assert.IsTrue(cts.IsCancellationRequested);
+            bool hasBeenCancelled = false;
+            // check if OperationCanceledException is in the Exception list
+            // may not be first one in async case
+            foreach (var exception in exceptions.InnerExceptions)
```

Comment:
Could we use a linq expression instead with [Any](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.any?view=net-7.0)?